code-challenges.js

```diff
@@ -21,44 +32,92 @@ var num3 = -7
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
 
+const capitalizeWords = wordsArrayIn =>
+{ return wordsArrayIn.map(w => w = w.charAt(0).toUpperCase() + w.substr(1)) }
 
 
+log(capitalizeWords(randomNouns))
+
 
 
 // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
 
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+const grabAndSortNumbers = arrMiscIn =>
+{ return arrMiscIn.filter(el => typeof el === "number" ).sort((a, b) => a - b) }
+
 
+log(grabAndSortNumbers(mixedDataArray))
 
 
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
 // Use the test variables provided.
 
-var vowelTester1 = "learn"
-// Expected output: 1
-var vowelTester1 = "throw"
-// Expected output: 3
+var vowelTester1 = "learn"     // Expected output: 1
+var vowelTester2 = "throw"     // Expected output: 3
 
+const logFirstIndexOfVowel = wordIn => {
 
+    let vowels = ["a","e","i","o","u"]
 
+    for(let i=0; i<wordIn.length; i++){
+        for (let vi=0; vi<vowels.length; vi++){
+            if(wordIn.charAt(i) === vowels[vi]) { return i }
+        }
+    }
+    return -1
```

Comment:
this code totally works! an extra challenge might be utilizing `.includes()` in this code instead of a nested loop. just a suggestion!